pkg/webservice/handlers_test.go

```diff
@@ -1358,17 +1364,42 @@ func TestGetPartitionNodes(t *testing.T) {
 	assertParamsMissing(t, resp)
 
 	// Test specific node
-	req, err = createRequest(t, "/ws/v1/partition/default/node/node-1", map[string]string{"node": "node-1"})
+	req, err = createRequest(t, "/ws/v1/partition/default/node/node-1", map[string]string{"partition": "default", "node": "node-1"})
 	assert.NilError(t, err, "Get Node for PartitionNode Handler request failed")
 	resp = &MockResponseWriter{}
 	getPartitionNode(resp, req)
+	var nodeInfo dao.NodeDAOInfo
+	err = json.Unmarshal(resp.outputBytes, &nodeInfo)
+	assert.NilError(t, err, unmarshalError)
+	assertNodeInfo(t, &nodeInfo, node1ID, "alloc-1", attributesOfnode1, map[string]int64{"memory": 50, "vcore": 30})
 
 	// Test node id is missing
 	req, err = createRequest(t, "/ws/v1/partition/default/node/node-1", map[string]string{"partition": "default", "node": ""})
 	assert.NilError(t, err, "Get Node for PartitionNode Handler request failed")
 	resp = &MockResponseWriter{}
 	getPartitionNode(resp, req)
 	assertNodeIDNotExists(t, resp)
+
+	// Test param missing
+	req, err = http.NewRequest("GET", "/ws/v1/partition/default/node", strings.NewReader(""))
+	assert.NilError(t, err, "Get Node for PartitionNode Handler request failed")
+	resp = &MockResponseWriter{}
+	getPartitionNode(resp, req)
+	assertParamsMissing(t, resp)
+
+	// Test partition not exist
+	req, err = createRequest(t, "/ws/v1/partition/notexists/node/node-1", map[string]string{"partition": "notexists"})
+	assert.NilError(t, err, "Get Nodes for PartitionNode Handler request failed")
+	resp = &MockResponseWriter{}
+	getPartitionNodes(resp, req)
+	assertPartitionNotExists(t, resp)
+}
```

Comment:
It is growing big. Probably, time to split into two separate methods, one for Nodes and another one for specific node.

Reply:
Done.